doc/api/async_hooks.md

```diff
@@ -0,0 +1,677 @@
+  },
+  destroy (id) {
+    process._rawDebug(' '.repeat(ws) + 'destroy:', id);
+  },
+}).enable();
+
+require('net').createServer(() => {}).listen(8080, () => {
+  // Let's wait 10ms before logging the server started.
+  setTimeout(() => {
+    console.log('>>>', async_hooks.currentId());
+  }, 10);
+});
+```
+
+Output from only starting the server:
+
+```
+TCPWRAP(2): trigger: 1 scope: 1
+TickObject(3): trigger: 2 scope: 1
+before:  3
+  Timeout(4): trigger: 3 scope: 3
+  TIMERWRAP(5): trigger: 3 scope: 3
+after:   3
+destroy: 3
+before:  5
+  before:  4
+    TTYWRAP(6): trigger: 4 scope: 4
+    SIGNALWRAP(7): trigger: 4 scope: 4
+    TTYWRAP(8): trigger: 4 scope: 4
+>>> 4
+    TickObject(9): trigger: 4 scope: 4
+  after:   4
+  destroy: 4
+after:   5
+before:  9
+after:   9
+destroy: 9
+destroy: 5
+```
+
+First notice that `scope` and the value returned by `currentId()` are always
+the same. That's because `currentId()` simply returns the value of the
+current execution context; which is defined by `before()` and `after()` calls.
+
+Now only using `scope` to graph resource allocation results in the following:
+
+```
+TTYWRAP(6) -> Timeout(4) -> TIMERWRAP(5) -> TickObject(3) -> root(1)
+```
+
+The `TCPWRAP` isn't part of this graph; even though it was the reason for
+`console.log()` being called. This is because binding to a port without a
+hostname is actually synchronous, but to maintain a completely asynchronous API
+the user's callback is placed in a `process.nextTick()`.
+
+The graph only shows **when** a resource was created. Not **why**. So to track
+the **why** use `triggerId`.
+
+
+##### `before(id)`
+
+* `id` {number}
+
+When an asynchronous operation is triggered (such as a TCP server receiving a
+new connection) or completes (such as writing data to disk) a callback is
+called to notify Node. The `before()` callback is called just before said
+callback is executed. `id` is the unique identifier assigned to the
+resource about to execute the callback.
+
+The `before()` callback will be called 0-N times if the resource is a handle,
+and exactly 1 time if the resource is a request.
+
+
+##### `after(id)`
+
+* `id` {number}
+
+Called immediately after the callback specified in `before()` is completed. If
+an uncaught exception occurs during execution of the callback then `after()`
+will run after the `'uncaughtException'` event or a `domain`'s handler runs.
+
+
+##### `destroy(id)`
+
+* `id` {number}
+
+Called either when the class destructor is run or if the resource is manually
+marked as free. For core C++ classes that have a destructor the callback will
+fire during deconstruction. It is also called synchronously from the embedder
+API `emitDestroy()`.
+
+Some resources, such as `HTTPParser`, are not actually destructed but instead
+placed in an unused resource pool to be used later. For these `destroy()` will
+be called just before the resource is placed on the unused resource pool.
```

Comment:
I’d drop this (again, unless it’s visible to outside observers, but I’m pretty sure it isn’t).